apps/wpcom-block-editor/src/wpcom/features/tracking.js

```diff
@@ -84,14 +84,34 @@ const trackBlockReplacement = ( originalBlockIds, blocks ) => {
  * @param {object|Array} blocks block instance object or an array of such objects
  * @returns {void}
  */
-const trackInnerBlocksReplacement = ( rootClientId, blocks ) => {
-	castArray( blocks ).forEach( block => {
-		tracksRecordEvent( 'wpcom_block_inserted', {
-			block_name: block.name,
-			blocks_replaced: false,
-			// isInsertingPageTemplate filter is set by Starter Page Templates
-			from_template_selector: applyFilters( 'isInsertingPageTemplate', false ),
-		} );
+let blockCounter = 0;
+const trackInnerBlocksReplacement = ( rootClientId = '', blocks ) => {
+	const castBlocks = castArray( blocks );
+	if ( ! castBlocks || ! castBlocks.length ) {
+		return;
+	}
+
+	castBlocks.forEach( block => {
+		setTimeout(
```

Comment:
Maybe you are trying to defer the calls to avoid blocking the editor from rendering? If so then this will push each call onto the next "tick" of the Event loop:

```js
setTimeout(fn, 0)
```

Guess it's much the same. Either way we should comment this to make sure it's clear what's going on.

> This method can be used to break up long running operations and run a callback function immediately after the browser has completed other operations such as events and display updates.

Reply:
Yeah.
First of all, probably it doesn't make sense.

One reason to delay tracking these events is avoiding to affect with the template rendering process. A template could contain hundreds of blocks and waiting to track all events could affect it in terms of performance. Again, not sure about this.
My other concern is triggering these requests at the same time. For this reason, I've delayed, one by one, every tacking event. I think that 200 requests at the same time could be something undesired, but definitely it worths research and confirm this fragile theory.
Anyway, removing this will get the approach easier and simpler. Glad to do it.